src/js_printer.zig

```diff
@@ -4936,10 +4936,19 @@ fn NewPrinter(
                             p.printIndent();
                         }
                     }
+
+                    if (s.yes.isMissingExpr()) {
+                        p.printSemicolonIfNeeded();
+                    }
                 },
             }
 
             if (s.no) |no_block| {
+                if (no_block.isMissingExpr() and p.options.minify_syntax) {
```

Comment:
are you sure we can't fix this in the parser so that this never happens in the first place?

Reply:
moved the fix to the parser